install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -131,9 +131,9 @@ data:
   skip-cnp-status-startup-clean: "{{ .Values.operator.skipCNPStatusStartupClean }}"
 {{- end }}
 
-{{- if hasKey .Values "disableEndpointCRD" }}
+{{- if .Values.disableEndpointCRD }}
```

Comment:
to enforce an explicit type check - we have to change this to `{{- if eq .Values.disableEndpointCRD true }}`

this will result in an error if trying to pass an incompatible type.

```
error calling eq: incompatible types for comparison
```

Reply:
Updated and tested.

Configuration with the wrong type
```
enableCiliumEndpointSlice: false
disableEndpointCRD: "true"
```
Error
```
Error: template: cilium/templates/cilium-configmap.yaml:134:7: executing "cilium/templates/cilium-configmap.yaml" at <eq .Values.disableEndpointCRD true>: error calling eq: incompatible types for comparison
```

Configuration with the wrong type BUT enableCiliumEndpointSlice: true
```
enableCiliumEndpointSlice: true
disableEndpointCRD: "true"
```
The error type is not thrown since I guess validate.yaml is the first test to be executed.
```
Error: execution error at (cilium/templates/validate.yaml:76:7): if Cilium Endpoint Slice is enabled (.Values.enableCiliumEndpointSlice=true), it requires .Values.disableEndpointCRD=false
```

Reply:
@doniacld great

we're missing one case:

```
enableCiliumEndpointSlice: true
disableEndpointCRD: "false"
```

the validation error gets thrown even though the endpointCRD isn't actually disabled 

```
❯ helm template ./install/kubernetes/cilium -f ./contrib/testing/kind-values.yaml
Error: execution error at (cilium/templates/validate.yaml:76:7): if Cilium Endpoint Slice is enabled (.Values.enableCiliumEndpointSlice=true), it requires .Values.disableEndpointCRD=false
```

-> we should change the validation to enforce type checks too

```
{{- if and (eq .Values.enableCiliumEndpointSlice true ) (eq .Values.disableEndpointCRD true ) }}
  {{ fail "if Cilium Endpoint Slice is enabled (.Values.enableCiliumEndpointSlice=true), it requires .Values.disableEndpointCRD=false" }}
{{- end }}
```

Reply:
thanks @doniacld